.github/workflows/release.yml

```diff
@@ -3,16 +3,90 @@ name: Release
 
 on:
   pull_request:
-    types: closed
-
+    types: [ closed ]
 jobs:
   build:
     runs-on: ubuntu-latest
+    permissions: write-all
+    if: github.event.pull_request.merged
```

Comment:
Assuming this adds an extra condition so it only runs on closed pull requests that were merged, and not just any closed PR. Cool